src/components/resource-opening-hours/ResourceOpeningHours.tsx

```diff
@@ -80,6 +86,47 @@ const ResourceOpeningHours = ({
     (datePeriod) => isHolidayOrEve(datePeriod, holidays)
   );
 
+  const allDatePeriodsSelected =
+    selectedDatePeriods.length ===
+      normalDatePeriods.length + exceptions.length &&
+    selectedDatePeriods.length > 0;
+
+  const onChangeHandler = (): void => {
+    if (!allDatePeriodsSelected) {
+      const allDatePeriods = [...normalDatePeriods, ...exceptions];
+      addDatePeriods(allDatePeriods);
+    } else {
+      clearDatePeriods();
+    }
+  };
+
+  // this is the ONLY place where we still have the information about the datePeriod type!
+  useEffect(() => {
+    const selectedNormalDatePeriods: DatePeriod[] = normalDatePeriods
+      .filter((dp) => selectedDatePeriods.some((d) => d.id === dp.id))
+      .map((dp) => ({ ...dp, type: 'NORMAL' }));
+    const selectedExceptionDatePeriods: DatePeriod[] = exceptionDatePeriods
+      .filter((dp) => selectedDatePeriods.some((d) => d.id === dp.id))
+      .map((dp) => ({ ...dp, type: 'EXCEPTION' }));
+    const selectedHolidaynDatePeriods: DatePeriod[] = holidayDatePeriods
+      .filter((dp) => selectedDatePeriods.some((d) => d.id === dp.id))
```

Comment:
Hmm... I tried but ended up with more code than this :D because we need to filter 3 different arrays of datePeriods from selected periods and add the correct type to them 🤷 But if you know a better way, I'm all 👂 s @mikkojamG

Reply:
I had in mind something like this 🤔 :
```
const filterDatePeriods = useCallback(
    (datePeriods: DatePeriod[], type: DatePeriodType) => {
      return datePeriods
        .filter((dp) => selectedDatePeriods.some((d) => d.id === dp.id))
        .map((dp) => ({ ...dp, type }));
    },
    [selectedDatePeriods]
  );
  
...
  
  // this is the ONLY place where we still have the information about the datePeriod type!
  useEffect(() => {
    const selectedNormalDatePeriods = filterDatePeriods(
      normalDatePeriods,
      DatePeriodType.NORMAL
    );
...
```

Reply:
Yeah just what I started writing a moment ago... dunno which is still better? 🤷 still 3 lines of code per type inside the useEffect plus the separate function 🤷

Reply:
I would argue that it makes it easier to read 😁 But you are right, doesn't make it shorter.. 👀